examples/aspnetcore20-mvc/Controllers/HomeController.cs

```diff
@@ -9,29 +9,71 @@
 
 namespace aspnetcore20_mvc.Controllers
 {
-    public class HomeController : Controller
+    public class HomeController : Microsoft.AspNetCore.Mvc.Controller
     {
-        private readonly IClient _client;
+        // The Bugsnag client (initialized in Startup.cs) will be injected into your classes where you declare the IClient dependency.
+        // This allows you to report handled exceptions within your controller.
+        private readonly Bugsnag.IClient _bugsnag;
 
-        public HomeController(IClient client)
+        public HomeController(Bugsnag.IClient client)
         {
-          _client = client;
+            _bugsnag = client;
+            // A BeforeNotify callback lets you evaluate, modify, add and remove data before sending the error to bugsnag. The actions here will be applied to *all* errors, handled and unhandled.
+            _bugsnag.BeforeNotify(report => {
+                // In order to correlate errors with customer reports, or to see a list of users who experienced each error, you can attach user data in your callback
+                report.Event.User = new Bugsnag.Payload.User
+                {
+                    Id = "006",
+                    Name = "Hedy Lamarr",
+                    Email = "h.lamarr@ww.com"
+                };
+                //this example makes some modifications that only apply to reports of error class "System.NotImplementedException".
+                if (report.OriginalException is System.NotImplementedException)
+                {
+                    report.Event.Metadata.Add("account", new Dictionary<string, object> { { "paying", true } });
+                    report.Event.Context = "an-important-context";
+                }
+
+                // note that if you return false from the BeforeNotify, this will cancel the entire error report.
```

Comment:
this is not true, you need to call `report.Ignore()` to stop the report from sending